docs/guide/mocking-requests.md

```diff
@@ -0,0 +1,57 @@
+# Mocking Requests
+
+We write tests in order to be able to ship with confidence. The more underlying behavior we mock, the less confidence we get since the system is less closer to real usage.
+
+Mocking network requests is one of those tricky aspects of running tests in a non-browser environment. Web API's are not available so now we need to mock or install something that mocks this behavior.
+
+There are various tools for this job, but which one do you choose?
+
+We recommend [Mock Service Worker](https://mswjs.io/) to mock network requests. It handles both `REST` and `GraphQL` type of network requests and is framework agnostic.
+
+Mock Service Worker (or MSW in short) uses [Service Worker API](https://developers.google.com/web/fundamentals/primers/service-workers) to intercept requests. It has a variety of use cases but also offers functionality specifically for `node`. It won't magically make `service workers API` available for non-browser environments, but instead relies on [node-request-interceptor](https://mswjs.io/docs/api/setup-server#operation).
+
+To read more about MSW, read their [introduction](https://mswjs.io/docs/).
+
+## Configuration
+
+Add the following to your (test) setup file
+```js
+import { beforeAll, afterAll, afterEach } from 'vitest';
```

Comment:
```suggestion
import { beforeAll, afterAll, afterEach } from 'vitest'
```

While I prefer trailing semicolons for everything, the rest of this doc doesn't have them. We should have consistency throughout the doc.